### DISCUSSION TITLE
DB netbox broken after upgrade 3.4

### DISCUSSION THREAD

**Discussion by marcucci-DAE** [netbox-community/netbox#11293] [Q&A]
### NetBox version

v3.4.1

### Python version

3.9

### Steps to Reproduce

After running the ./upgrade.sh (as usually I do..) In the phase the manage.py migrate I receive this error .

django.db.utils.IntegrityError: could not create unique index "dcim_device_unique_name_site_tenant"
DETAIL:  Key (lower(name::text), site_id, tenant_id)=(ap-**-n**, 1, 1) is duplicated.

And the DB is not accessible from netbox...

Please help me

### Expected Behavior

django.db.utils.IntegrityError: could not create unique index "dcim_device_unique_name_site_tenant"
DETAIL:  Key (lower(name::text), site_id, tenant_id)=(ap-**-n**, 1, 1) is duplicated.

### Observed Behavior

django.db.utils.IntegrityError: could not create unique index "dcim_device_unique_name_site_tenant"
DETAIL:  Key (lower(name::text), site_id, tenant_id)=(ap-**-n**, 1, 1) is duplicated.

**Comment by renatoalmeidaoliveira** [comment]
It seems that you might have devices with the same name in some site, in version 3.4 the name is case unsensitive so:
Device == device.
Are you able to run nbshell to change the devices name, or rollback to 3.3 change the names and upgrade again?

**Comment by marcucci-DAE** [comment]
Thanks for your reply . How I can modify thesw device into db?

Thanks

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________
From: Renato Almeida de Oliveira ***@***.***>
Sent: Tuesday, December 20, 2022 7:54:30 PM
To: netbox-community/netbox ***@***.***>
Cc: Tiziana Marcucci ***@***.***>; Author ***@***.***>
Subject: [EXT] Re: [netbox-community/netbox] DB netbox broken after upgrade 3.4 (Issue #11266)


ATTENZIONE: Questa email ha origine esterna all’organizzazione. Non aprire i link e MAI scaricare gli allegati, a meno di conoscere il mittente e di sapere che il contenuto sia attendibile.
WARNING: This mail source is external to the organization. Do not open any link and NEVER download attachments, unless you trust the sender and its contents.

It seems that you might have devices with the same name in some site, in version 3.4 the name is case unsensitive so:
Device == device.

**Comment by renatoalmeidaoliveira** [comment]
Are you able to access nbshell?

**Comment by marcucci-DAE** [comment]
Yes of course.

Da: Renato Almeida de Oliveira ***@***.***>
Inviato: martedì 20 dicembre 2022 20:08
A: netbox-community/netbox ***@***.***>
Cc: Tiziana Marcucci ***@***.***>; Author ***@***.***>
Oggetto: [EXT] Re: [netbox-community/netbox] DB netbox broken after upgrade 3.4 (Issue #11266)

ATTENZIONE: Questa email ha origine esterna all’organizzazione. Non aprire i link e MAI scaricare gli allegati, a meno di conoscere il mittente e di sapere che il contenuto sia attendibile.
WARNING: This mail source is external to the organization. Do not open any link and NEVER download attachments, unless you trust the sender and its contents.

Are you able to access nbshell?

**Comment by renatoalmeidaoliveira** [comment]
You have two options here, export all your duplicated names to a csv and deal with then by script later, or append an index to each of then and deal in the web interface. Which one you prefer?

**Comment by marcucci-DAE** [comment]
The first tnks

Da: Renato Almeida de Oliveira ***@***.***>
Inviato: giovedì 22 dicembre 2022 14:56
A: netbox-community/netbox ***@***.***>
Cc: Tiziana Marcucci ***@***.***>; Author ***@***.***>
Oggetto: [EXT] Re: [netbox-community/netbox] DB netbox broken after upgrade 3.4 (Discussion #11293)

ATTENZIONE: Questa email ha origine esterna all’organizzazione. Non aprire i link e MAI scaricare gli allegati, a meno di conoscere il mittente e di sapere che il contenuto sia attendibile.
WARNING: This mail source is external to the organization. Do not open any link and NEVER download attachments, unless you trust the sender and its contents.

You have two options here, export all your duplicated names to a csv and deal with then by script later, or append an index to each of then and deal in the web interface. Which one you prefer?

**Comment by marcucci-DAE** [comment]
Hi,
 i prefer following the first solution, Could you tell me how i do it?

Thanks
Tiziana
Da: Renato Almeida de Oliveira ***@***.***>
Inviato: giovedì 22 dicembre 2022 14:56
A: netbox-community/netbox ***@***.***>
Cc: Tiziana Marcucci ***@***.***>; Author ***@***.***>
Oggetto: [EXT] Re: [netbox-community/netbox] DB netbox broken after upgrade 3.4 (Discussion #11293)

ATTENZIONE: Questa email ha origine esterna all’organizzazione. Non aprire i link e MAI scaricare gli allegati, a meno di conoscere il mittente e di sapere che il contenuto sia attendibile.
WARNING: This mail source is external to the organization. Do not open any link and NEVER download attachments, unless you trust the sender and its contents.

You have two options here, export all your duplicated names to a csv and deal with then by script later, or append an index to each of then and deal in the web interface. Which one you prefer?

**Comment by marcucci-DAE** [comment]
I have resolved this issue deleting manually the records duplicate
